File: 0045-jump-game-ii/0045-jump-game-ii.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        n = len(nums)
        
        dp = [inf] * n
        
        dp[0] = 0
        
        for i in range(n):
            for j in range(i+1, i+nums[i]+1):
                if j < n:
                    dp[j] = min(dp[j], 1 + dp[i])
        
        return dp[n-1]

File: 0045-jump-game-ii/0045-jump-game-ii.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var jump = function(nums) {
    const n = nums.length
    const dp = Array(n).fill(Infinity)
    dp[0] = 0
    for (let i=0; i < n; i++) {
        for (let j=i+1; j < n && j <= i + nums[i]; j++) {
            dp[j] = Math.min(dp[j], 1 + dp[i])
        }
    }
    return dp[n-1]
};